builder/builder.go

```diff
@@ -2091,14 +2143,14 @@ func writeMetaFiles(path, format, version string) error {
 	return ioutil.WriteFile(filepath.Join(path, "mixer-src-version"), []byte(version), 0644)
 }
 
-func (b *Builder) getUpstreamFormatRange() (format string, first, latest uint32, err error) {
-	format, err = b.DownloadFileFromUpstream(fmt.Sprintf("update/%d/format", b.UpstreamVerUint32))
+func (b *Builder) getUpstreamFormatRange(version string) (format string, first, latest uint32, err error) {
+	format, err = b.DownloadFileFromUpstreamAsString(fmt.Sprintf("update/%s/format", version))
```

Comment:
I think it'd be useful to keep the original functionality and check if version == "", then use b.UpstreamVerUint32 by default.

Reply:
Hmm, I'm not sure I understand the benefit yet. The original functionality was not having to pass anything, and it only knew how to work on the "current" version. If the calling sites have to pass something either way, I don't see why it's bad for them to pass the version they want. (I've also just never been a fan of passing empty string as an argument to things as a way of imitating optional fields.)

Reply:
Ok I can agree with that answer.